pkg/haproxy/haproxy_test.go

```diff
@@ -113,6 +113,19 @@ func TestStatefulset(t *testing.T) {
 		assert.Equal(t, runtimeClassName, *sts.Spec.Template.Spec.RuntimeClassName)
 	})
 
+	t.Run("service account name", func(t *testing.T) {
+		cluster := cr.DeepCopy()
+		sts := StatefulSet(cluster, initImage, configHash, tlsHash, secret)
+		var e string
+		assert.Equal(t, e, sts.Spec.Template.Spec.ServiceAccountName)
```

Comment:
Here we can do `assert.Empty(t, sts.Spec.Template.Spec.ServiceAccountName)` instead.